src/popup/components/layout/MainLayout.tsx

```diff
@@ -1,20 +1,44 @@
+import browser from 'webextension-polyfill'
 import React from 'react'
 import { Outlet } from 'react-router-dom'
 
+import { PERMISSION_HOSTS } from '@/shared/constants'
+
 import { Header } from './header'
+import { MissingHostPermission } from '@/popup/components/MissingHostPermission'
 
 const Divider = () => {
   return <div className="w-100 h-1 bg-divider-gradient" />
 }
 
 export const MainLayout = () => {
+  const [hasHostPermission, setHasHostPermission] = React.useState(true)
+
+  React.useEffect(() => {
```

Comment:
I'm not sure we should do these checks in popup or pass these through background (popup state). @dianafulga?

Reply:
I would pass it through the background, because:
1. we might want to add in the feature logic for a certain icon to be displayed
2. I would keep the browser usage from polyfill limited to background, to easily change it in the feature. This is why we actually inject it in the container in the first place

Reply:
> keep the browser usage from polyfill limited to background

Though, we'd eventually need it in popup, for example `browser.i18n.getMessage`, as well as for requesting permission (on user click). Maybe I can remove the polyfill there and add a lighter wrapper for these two at least. We don't need most of the polyfill there.